src/main/kotlin/io/libp2p/security/noise/NoiseXXSecureChannel.kt

```diff
@@ -0,0 +1,315 @@
+
+                        logger.debug("Reporting secure channel initialized")
+                    }
+                    is SecureChannelFailed -> {
+                        ret.completeExceptionally(evt.exception)
+
+                        ctx.pipeline().remove(handshakeHandlerName + chid)
+                        ctx.pipeline().remove(this)
+
+                        logger.debug("Reporting secure channel failed")
+                    }
+                }
+                ctx.fireUserEventTriggered(evt)
+                ctx.fireChannelActive()
+            }
+        }
+        ch.nettyChannel.pipeline().addLast(handshakeHandlerName + chid, NoiseIoHandshake())
+        ch.nettyChannel.pipeline().addLast(handshakeHandlerName + chid + "ResultHandler", resultHandler)
+        return ret
+    }
+
+    inner class NoiseIoHandshake() : SimpleChannelInboundHandler<ByteBuf>() {
+
+        private val handshakestate: HandshakeState = HandshakeState(protocolName, role.get())
+        private var loggerName: String
+        private var logger2: Logger
+
+        private var localNoiseState: DHState
+        private var sentNoiseKeyPayload = false
+
+        private val instancePayload = ByteArray(65535)
+        private var instancePayloadLength = 0
+
+        init {
+            val roleString = if (role.get() == HandshakeState.INITIATOR) "INIT" else "RESP"
+//            loggerName = loggerNameParent + "." + this.hashCode().toString().substring(5)
+            loggerName = roleString + "|" + chid + "|" + loggerNameParent
+            loggerName = loggerName.replace(".", "_")
+//            System.out.println("loggerName:"+loggerName)
+
+            logger2 = LogManager.getLogger(loggerName)
+            Configurator.setLevel(loggerName, Level.DEBUG)
+
+            logger2.debug("Starting handshake")
+
+            // configure the localDHState with the private
+            // which will automatically generate the corresponding public key
+            localNoiseState = Noise.createDH("25519")
+            localNoiseState.setPrivateKey(localStaticPrivateKey25519, 0)
+            handshakestate.localKeyPair.copyFrom(localNoiseState)
+            handshakestate.start()
+        }
+
+        override fun channelRead0(ctx: ChannelHandlerContext, msg1: ByteBuf) {
+            val msg = msg1.toByteArray()
+
+            channelActive(ctx)
+
+            if (role.get() == HandshakeState.RESPONDER && flagRemoteVerified && !flagRemoteVerifiedPassed) {
+                logger2.error("Responder verification of Remote peer id has failed")
+                ctx.fireUserEventTriggered(SecureChannelFailed(Exception("Responder verification of Remote peer id has failed")))
+                return
+            }
+
+            // we always read from the wire when it's the next action to take
+            // capture any payloads
+            val payload = ByteArray(65535)
```

Comment:
Can we avoid creating max-length buffer here and the similar cases below?

Reply:
Done.